modules/geoip/geoip-parser.c

```diff
@@ -24,24 +24,25 @@
 #include "geoip-parser.h"
 #include "parser/parser-expr.h"
 #include "scratch-buffers2.h"
+#include "geoip-helper.h"
 
-#include <GeoIPCity.h>
+typedef struct _GeoIPParser GeoIPParser;
 
-typedef struct
+struct _GeoIPParser
 {
   LogParser super;
   GeoIP *gi;
 
   gchar *database;
   gchar *prefix;
-
+  void (*lookup)(GeoIPParser *, LogMessage *, const gchar *);
   struct
```

Comment:
I like that this has become a virtual function. "lookup" as a name is not the best one though, as it not only looks up (which suggests a read-only function without side effects), it actually changes the message, thus has side effects.

"process" as a term is a bit too generic and is already used by "LogParser", so another term would be better. I am out of really good ideas, but these come to my mind: process_record(), process_geoip(), lookup_and_set_values(), but I think you get the idea.